### PR TITLE
Issue268 xsd versions

### PR DESCRIPTION
**Pull request by PRemmen**

### CODE REVIEW

File: teaser/data/bindings/v_0_3_9/material_bind.py [hunk 1]
```diff
@@ -147,42 +147,50 @@ class MaterialTemplatesType (pyxb.binding.basis.complexTypeDefinition):
     _ContentTypeTag = pyxb.binding.basis.complexTypeDefinition._CT_ELEMENT_ONLY
     _Abstract = False
     _ExpandedName = pyxb.namespace.ExpandedName(Namespace, 'MaterialTemplatesType')
-    _XSDLocation = pyxb.utils.utility.Location('D:\\PyXB-1.2.4\\scripts\\MaterialTemplates.xsd', 14, 2)
+    _XSDLocation = pyxb.utils.utility.Location('D:\\GIT\\pyxb\\scripts\\MaterialTemplates.xsd', 14, 2)
```
Comment (mlauster) [inline teaser/data/bindings/v_0_3_9/material_bind.py]: Is this intentionally?

File: teaser/data/bindings/v_0_3_9/material_bind.py [hunk 2]
```diff
@@ -191,110 +199,72 @@ def _BuildAutomaton ():
     import pyxb.utils.fac as fac
 
     counters = set()
-    cc_0 = fac.CounterCondition(min=0, max=1, metadata=pyxb.utils.utility.Location('D:\\PyXB-1.2.4\\scripts\\MaterialTemplates.xsd', 6, 6))
+    cc_0 = fac.CounterCondition(min=0, max=1, metadata=pyxb.utils.utility.Location('D:\\GIT\\pyxb\\scripts\\MaterialTemplates.xsd', 10, 6))
```
Comment (mlauster) [inline teaser/data/bindings/v_0_3_9/material_bind.py]: same here

File: teaser/data/bindings/v_0_3_9/material_bind.py [hunk 3]
```diff
@@ -191,110 +199,72 @@ def _BuildAutomaton ():
...
     counters.add(cc_0)
-    cc_1 = fac.CounterCondition(min=0, max=1, metadata=pyxb.utils.utility.Location('D:\\PyXB-1.2.4\\scripts\\MaterialTemplates.xsd', 7, 6))
+    cc_1 = fac.CounterCondition(min=0, max=1, metadata=pyxb.utils.utility.Location('D:\\GIT\\pyxb\\scripts\\MaterialTemplates.xsd', 11, 6))
```
Comment (mlauster) [inline teaser/data/bindings/v_0_3_9/material_bind.py]: same here

File: teaser/data/bindings/v_0_3_9/material_bind.py [hunk 4]
```diff
@@ -303,12 +273,12 @@ def _BuildAutomaton_ ():
     import pyxb.utils.fac as fac
 
     counters = set()
-    cc_0 = fac.CounterCondition(min=0, max=None, metadata=pyxb.utils.utility.Location('D:\\PyXB-1.2.4\\scripts\\MaterialTemplates.xsd', 15, 4))
+    cc_0 = fac.CounterCondition(min=0, max=None, metadata=pyxb.utils.utility.Location('D:\\GIT\\pyxb\\scripts\\MaterialTemplates.xsd', 16, 1))
```
Comment (mlauster) [inline teaser/data/bindings/v_0_3_9/material_bind.py]: same here

**Comment by PRemmen** [comment]
These are internal properties  Of pyxb indicating where the schema location was stored. They shouldn't affect usage. I just copied the xsds in that folder but they are exactly the same as you find in this repository!

File: teaser/data/dataclass.py [hunk 5]
```diff
@@ -41,23 +40,52 @@ class DataClass(object):
...
+        version_parse = element_tree.parse(self.path_tb)
+        __xml_file_tb = open(self.path_tb, 'r', encoding='utf-8')
+
+        if bool(version_parse.getroot().attrib) is False:
+            warnings.warn("You are using an old version of type building element data base XML file")
+            import teaser.data.bindings.v_0_3_9.typeelement_bind as tb_bind
+            self.element_bind = tb_bind.CreateFromDocument(__xml_file_tb.read())
+        elif version_parse.getroot().attrib['version'] == "0.3.9":
+            import teaser.data.bindings.v_0_3_9.typeelement_bind as tb_bind
+            self.element_bind = tb_bind.CreateFromDocument(__xml_file_tb.read())
```
Comment (mlauster) [inline teaser/data/dataclass.py]: Could this be simplified by:
`if bool(...):
  warnings.warn(...)
import teaser....
self. ...
`

Or do you have this structure to prepare further version with a different syntax?
This applies for a lot more occurences

Comment (PRemmen) [inline teaser/data/dataclass.py]: exactly, this is for other versions if you have a nicer way, go ahead!

I'm just preparing version 0.4:

´´´python
        elif version_parse.getroot().attrib['version'] == "0.3.9":
            import teaser.data.bindings.v_0_3_9.material_bind as mat_bind
            self.material_bind = mat_bind.CreateFromDocument(__xml_file_mat.read())
        elif version_parse.getroot().attrib['version'] == "0.4":
            import teaser.data.bindings.v_0_4.material_bind as mat_bind
            self.material_bind = mat_bind.CreateFromDocument(__xml_file_mat.read())
```

Comment (mlauster) [inline teaser/data/dataclass.py]: No, then it's fine :-)

**Comment by mlauster** [comment]
this closes #313 

@PRemmen, I solved #313 and modified call of function `open` in dataclass.py. Tests run in Python 3.4 and 2.7. Will you cross-check?

**Pull request merged**